### PR TITLE
Disable kubeadm for filebeat (both filbeat.input and filbeat.autodiscover)

### PR DESCRIPTION
**Pull request by gizas**
- Enhancement

~NOTE: Dont merge until https://github.com/elastic/elastic-agent-autodiscover/pull/98 is also merged~

## Proposed commit message

**WHAT:** Provide a configuration option to disable /api/v1/namespaces/kube-system/configmaps/kubeadm-config for **filebeat autodiscovery** and **add_kubernetes_metadata processor**. In more details user will disable the api requests  by using configuration use_kubeadm: false

In this PR we added the option in order to support both **filbeat.input** and **filebeat.autodiscover** blocks of configuration.

**WHY:** https://github.com/elastic/enhancements/issues/21172

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

1. Follow same steps as described https://github.com/elastic/beats/pull/40086
2. The filebeat configuration should contain only autodiscover config: https://github.com/elastic/beats/blob/main/deploy/kubernetes/filebeat-kubernetes.yaml#L133C1-L150C42

## Related issues

- Relates #https://github.com/elastic/beats/pull/40086 
- Relates #https://github.com/elastic/beats/issues/40276

## Screenshots

Screenshot 2024-09-10 at 11 27 21 AM

LOG processing continues successfully:

Screenshot 2024-09-10 at 11 51 22 AM

For the filebeat.input [block](https://github.com/elastic/beats/blob/main/deploy/kubernetes/filebeat-kubernetes.yaml):

Screenshot 2024-09-10 at 2 17 25 PM

LOG processing continues successfully:
Screenshot 2024-09-10 at 2 30 40 PM

### CODE REVIEW

File: dev-tools/kubernetes/filebeat/Dockerfile.debug [hunk 1]
```diff
@@ -4,7 +4,7 @@ ENV PATH=/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/go/bin:/usr/local/go/bin
 
 ENV CGO_ENABLED=0
 
-RUN go install github.com/go-delve/delve/cmd/dlv@v1.21.2
+RUN go install github.com/go-delve/delve/cmd/dlv@v1.23.0
```
Comment (gizas) [inline dev-tools/kubernetes/filebeat/Dockerfile.debug:7]: the dlv changes are unrelated to this PR. They are needed in order to make the debugger working with newer version of go

File: metricbeat/module/kubernetes/util/kubernetes.go [hunk 2]
```diff
@@ -760,8 +765,14 @@ func NewContainerMetadataEnricher(
 		return &nilEnricher{}
 	}
 	// We initialise the use_kubeadm variable based on modules KubeAdm base configuration
-	config.AddResourceMetadata.Namespace.SetBool("use_kubeadm", -1, commonMetaConfig.KubeAdm)
-	config.AddResourceMetadata.Node.SetBool("use_kubeadm", -1, commonMetaConfig.KubeAdm)
+	err = config.AddResourceMetadata.Namespace.SetBool("use_kubeadm", -1, commonMetaConfig.KubeAdm)
+	if err != nil {
```
Comment (gizas) [inline metricbeat/module/kubernetes/util/kubernetes.go:769]: Although metricbeat is not related with this PR, I added checks here as well in order to have the same style in all beats

File: libbeat/autodiscover/providers/kubernetes/pod.go [hunk 3]
```diff
@@ -97,6 +97,15 @@ func NewPodEventer(uuid uuid.UUID, cfg *conf.C, client k8s.Interface, publish fu
 	}
 
 	metaConf := config.AddResourceMetadata
+	// We initialise the use_kubeadm variable based on modules KubeAdm base configuration
+	err = metaConf.Namespace.SetBool("use_kubeadm", -1, config.KubeAdm)
+	if err != nil {
```
Comment (gizas) [inline libbeat/autodiscover/providers/kubernetes/pod.go:102]: Added due to go-lint errors

File: filebeat/_meta/test/docs/01_playground/filebeat.yaml [hunk 4]
```diff
@@ -9,26 +9,44 @@ metadata:
...
         - add_kubernetes_metadata:
+            #use_kubeadm: true
             host: ${NODE_NAME}
             matchers:
             - logs_path:
                 logs_path: "/var/log/containers/"
 
     # To enable hints based autodiscover, remove `filebeat.inputs` configuration and uncomment this:
-    #filebeat.autodiscover:
+    # filebeat.autodiscover:
```
Comment (belimawr) [inline filebeat/_meta/test/docs/01_playground/filebeat.yaml:32]: I believe it's mostly cosmetic, but it would be better to keep the indentation the same as the rest of this file
```suggestion
    #filebeat.autodiscover:
```

File: filebeat/_meta/test/docs/01_playground/filebeat.yaml [hunk 5]
```diff
@@ -9,26 +9,44 @@ metadata:
...
     #        paths:
-    #          - /var/log/containers/*${data.kubernetes.container.id}.log
+    #        - /var/log/containers/*-${data.kubernetes.container.id}.log
+    #        parsers:
+    #        - container: ~
+    #        prospector:
+    #         scanner:
+    #           fingerprint.enabled: true
+    #           symlinks: true
+    #        file_identity.fingerprint: ~
```
Comment (belimawr) [inline filebeat/_meta/test/docs/01_playground/filebeat.yaml:49]: Same thing on this block, keep the indentation to two spaces

File: libbeat/processors/add_kubernetes_metadata/docs/add_kubernetes_metadata.asciidoc [hunk 6]
```diff
@@ -152,6 +152,7 @@ Example:
 -------------------------------------------------------------------------------------
 `kube_config`:: (Optional) Use given config file as configuration for Kubernetes
 client. It defaults to `KUBECONFIG` environment variable if present.
+`use_kubeadm`::  (Optional) Default to be true. By default requests to kubeadm config map are made in order to enrich cluster name by requesting /api/v1/namespaces/kube-system/configmaps/kubeadm-config API endpoint.
```
Comment (belimawr) [inline libbeat/processors/add_kubernetes_metadata/docs/add_kubernetes_metadata.asciidoc:155]: Small change in the text:
```suggestion
`use_kubeadm`::  (Optional) Default true. By default requests to kubeadm config map are made in order to enrich cluster name by requesting /api/v1/namespaces/kube-system/configmaps/kubeadm-config API endpoint.
```

File: libbeat/processors/add_kubernetes_metadata/kubernetes.go [hunk 7]
```diff
@@ -147,6 +147,18 @@ func (k *kubernetesAnnotator) init(config kubeAnnotatorConfig, cfg *config.C) {
 	k.initOnce.Do(func() {
 		var replicaSetWatcher, jobWatcher, namespaceWatcher, nodeWatcher kubernetes.Watcher
 
+		// We initialise the use_kubeadm variable based on modules KubeAdm base configuration
+		config.AddResourceMetadata.Namespace.SetBool("use_kubeadm", -1, config.KubeAdm)
```
Comment (MichaelKatsoulis) [inline libbeat/processors/add_kubernetes_metadata/kubernetes.go:151]: these two lines are not needed

**Review by MichaelKatsoulis — APPROVED** [review]

File: filebeat/_meta/test/docs/01_playground/filebeat.yaml (hunk 5 above)
Comment (gizas) [reply filebeat/_meta/test/docs/01_playground/filebeat.yaml:49]: done . Can you please have another look.
Thank you

**Review by leehinman — APPROVED** [review]

File: filebeat/_meta/config/filebeat.autodiscover.reference.yml.tmpl [hunk 8]
```diff
@@ -14,3 +14,26 @@
...
 #              paths:
 #                - /var/log/containers/*.log
+
+#Example: for kubernetes container logs autodiscovery
+#  filebeat.autodiscover:
+#    providers:
+#      - type: kubernetes
+#        node: ${NODE_NAME}
+#        hints.enabled: true
+#        use_kubeadm: true
```
Comment (tetianakravchenko) [inline filebeat/_meta/config/filebeat.autodiscover.reference.yml.tmpl:24]:
```suggestion
#        # By default requests to kubeadm config map are made in order to enrich cluster name by requesting /api/v1/namespaces/kube-system/configmaps/kubeadm-config API endpoint.
#        use_kubeadm: true
```

would it be better to move this description on top of the config parameter, instead of adding it to the end?
the same comments for other similar files

**Review by tetianakravchenko — APPROVED** [review]
